text/0015-binary-to-ternary-encoding/0015-binary-to-ternary-encoding.md

```diff
@@ -0,0 +1,79 @@
++ Feature name: `binary-to-ternary-encoding`
++ Start date: 2020-06-08
++ RFC PR: [iotaledger/protocol-rfcs#0015](https://github.com/iotaledger/protocol-rfcs/pull/15)
+
+# Summary
+
+In the IOTA protocol, a transaction is represented as ternary data. However, sometimes it is necessary to store binary data (e.g. the digest of a binary hash function) inside of a transaction. This requires the conversion of binary into ternary strings.
+The IOTA client libraries support the opposite conversion that encodes 5 trits as 1 byte (sometimes also referred to as `t5b1`), which is used for network communication and in storage layers. This RFC describes the corresponding counterpart to encode 1 byte as 6 trits.
+
+# Motivation
+
+A byte is composed of 8 bits that can represent 2<sup>8</sup> = 256 different values. On the other hand, 6 trits can hold 3<sup>6</sup> = 729 values while 5 trits can hold 3<sup>5</sup> = 243 values. Therefore, the most memory-efficient way to encode one byte requires the use of 6 trits. Although there exist many potential encoding schemes to convert binary data into ternary, the proposed version has been designed to directly match the widely used `t5b1` encoding.
```

Comment:
> Therefore, the most memory-efficient way to encode one byte requires the use of 6 trits.

1. That sentence is very true for one byte. But when I read it I paused and I started to think about what will happen if we want to convert more than one byte. For example if we try to take the number of bytes that make up a hash there is a more efficient encoding. Maybe we should mention it so future readers won't pause and ask questions?

2. Since we do `t5b1` it makes sense that we can do `b1t5`. In IRI there is a `Converter#getTrits` that does so using an internal mapping. I think (not sure) that bee has (or used to have) something like this as well.
I recall that somewhere you had discussions with @thibault-martinez about this. An explanation on the RFC would also be welcome :-)

Reply:
I think a bit more down in rational and alternatives you addressed (1)

Reply:
Made the difference between `t1b5` and `b1t6` clearer in an additional paragraph.
I hop that addresses this comment.